### DISCUSSION TITLE
VI Mode Escape Key is Frustrating

### DISCUSSION THREAD

**Discussion by wrynegade** [dbcli/pgcli#1340] [Q&A]
In `vi = True` mode, if a suggestion prompt is open the `[Esc]` key does not immediately exit insert mode. It seems like it tries to close the suggestion menu, but there is a 1 second delay between the key press and the menu finally closing. I'm always left frustratedly tapping the escape key over and over and over and never leaving insert mode!

When I press escape, I want to immediately be in normal mode regardless of whether a suggestion prompt is open. Is there a flag for this in the config that I am missing?

If not, could we make one? :)

**Comment by joshua-dean** [comment]
I couldn't find a config flag for this, so I suspect one would have to be made.

This appears to be behavior defined by `prompt_toolkit`.
Relevant issues: 
- https://github.com/prompt-toolkit/python-prompt-toolkit/issues/1901
- https://github.com/prompt-toolkit/python-prompt-toolkit/issues/1181

From the first issue, a valid solution appears to be setting `ttimeout_len` and `timeout_len` on the application object.
Adding this to the `cli()` function in main appears to remove the delay for me:
```python
    from prompt_toolkit.application import get_app
    get_app().timeoutlen = 0
    get_app().ttimeoutlen = 0
```

Here are links to those two options in the `prompt_toolkit` repo:
- https://github.com/prompt-toolkit/python-prompt-toolkit/blob/d8adbe9bfcf5f7e95b015d8ab12e2985b9a7822b/src/prompt_toolkit/application/application.py#L292
- https://github.com/prompt-toolkit/python-prompt-toolkit/blob/d8adbe9bfcf5f7e95b015d8ab12e2985b9a7822b/src/prompt_toolkit/application/application.py#L301

I think adding those as configuration options would be beneficial. The default delay is inconvenient.
